### PR TITLE
`scala-cli` script for development

### PR DESCRIPTION
**Pull request by tpasternak**
For scala-cli developers. Useful if you need a scala-cli exec that builds and runs code

### CODE REVIEW

**Review by lwronski — COMMENTED** [review]
In my opinion, we should use `standaloneLauncher`, it allows us to share `scala-cli` from the development build. 

We need also any mention about this script in Readme.

**Comment by tpasternak** [comment]
We don't need to merge it, it was just useful for me. I can keep it locally

**Comment by romanowski** [comment]
I like the idea! I would only rename the file to `scala-cli-mill` (or anything that would suggest that it will be run in jvm). Then, later we can add other script that along the lines of @lwronski idea

**Comment by tpasternak** [comment]
I wanted to name it `scala-cli` in order to provide the experience of a released `scala-cli` after just exporting PATH. It's useful for presentations, screenshots etc.

**Comment by alexarchambault** [comment]
We could put it at the root IMO, and name it… `scala-cli-src` for example?

And the current `scala-cli` script at the root of the repo could be removed IMO (unless anyone here uses it?). It's supposed to run the native image built in the sources (but only if it already exists, it doesn't build it itself).

**Comment by tpasternak** [comment]
TBH I didn't notice the old script in root when I was putting this. I preferred it not to bin in a root because the main idea behind this is to  run`export PATH=$PATH:$root/tools/bin` and have the same experience as using the original scala-cli (work example shebangs work then, or you can just copy the exact command from a bug report)

**Comment by alexarchambault** [comment]
It's fine to put it in the root IMO. Unless we have other such scripts lined up for a `tools/bin` directory?

File: scala-cli-src [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (alexarchambault) [inline scala-cli-src]: NIT
```suggestion
#!/usr/bin/env bash
```
(required on nix IIRC)

Comment (alexarchambault) [reply scala-cli-src]: Also, let's add at least `set -e` just in case:
```suggestion
#!/usr/bin/env bash
set -e
```

File: scala-cli-src [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+ROOT_DIR="$SCRIPT_DIR"
+(
+   cd "$ROOT_DIR" || exit
+   ./mill cli.standaloneLauncher
+)
+"$ROOT_DIR/out/cli/standaloneLauncher/dest/launcher" "$@"
```
Comment (alexarchambault) [inline scala-cli-src]: We can use `exec` here, which will make the `scala-cli-src` invocation exit with the scala-cli exit code.
```suggestion
exec "$ROOT_DIR/out/cli/standaloneLauncher/dest/launcher" "$@"
```

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**